### PR TITLE
Improve Cruft Remover UI

### PR DESCRIPTION
**Pull request by mikkoaf**
The user is given an option of selecting multiple items considered as cruft and removing from the server.
image
Choosing **Select All** chooses all the items on the list, including it's counterpart on the bottom of the list.
image
Deselecting any item from the list removes the **Select All** selection.
image
If all items are deleted, the Cruft remover returns to it's empty state.
image

### CODE REVIEW

File: js/cruftfiles.js [hunk 1]
```diff
@@ -33,29 +33,70 @@ jQuery(document).ready(function($) {
...
-            $( '#cruftfiles_entries' ).append('<tr class="cruftfile"><td class="cruftfile-delete"><a href="" class="dashicons dashicons-trash cruftfile-delete-button"></td><td class="cruftfile-path">'
+            $( '#cruftfiles_entries' ).append('<tr class="cruftfile"><td class="cruftfile-delete"><input data-file-name="' + file + '" class="cruftfile-check" type="checkbox"></td><td class="cruftfile-path">'
                                                 + file + '</td></tr>');
           }
         });
         if (filecount == 0) {
+
           $( '#cruftfiles_status' ).append('<b>' + seravo_cruftfiles_loc.no_cruftfiles + '</b>');
+        } else {
+          $( '#cruftfiles_entries' ).parents(':eq(0)').prepend('<thead><tr><td><input class="cruftfile-select-all" type="checkbox" ></td><td class="cruft-tool-selector"><b>Select all files</b></td></tr></thead>');
```
Comment (l3ku) [inline js/cruftfiles.js]: What is the purpose to have a delete button and select all boxes on both sides of the table? I think it looks rather crowded when there is e.g. only one cruft file:
screen shot 2018-06-28 at 9 43 13

If there are more cruft files laying around, it looks a bit better:
screen shot 2018-06-28 at 9 49 56

Of course our UI experts @nasusa and @tlxo  can better evaluate the design :)

File: js/cruftfiles.js [hunk 2]
```diff
@@ -33,29 +33,70 @@ jQuery(document).ready(function($) {
...
           event.preventDefault();
           var is_user_sure = confirm(seravo_cruftfiles_loc.confirm);
           if ( ! is_user_sure) {
             return;
           }
-          var parent_row = $(this).parents(':eq(1)');
-          var filepath = parent_row.find('.cruftfile-path').html();
+          var cruft_list = new Array();
+          var remove_rows = new Array();
+          //jokainen checkattu boxi - sen sisaruksess on tiedostonimi
```
Comment (l3ku) [inline js/cruftfiles.js]: Inline comments in English please.

File: js/cruftfiles.js [hunk 3]
```diff
@@ -33,29 +33,70 @@ jQuery(document).ready(function($) {
...
+          $('.cruftfile-check').each(function(){
+            if ( $(this).is(":checked") ) {
+              remove_rows.push( $(this).parents(':eq(1)') );
+              cruft_list.push( $(this).attr('data-file-name') );
+            }
+          });
+          //var parent_row = $(this).parents(':eq(1)');
+          //var filepath = parent_row.find('.cruftfile-path').html();
```
Comment (l3ku) [inline js/cruftfiles.js]: Remove commented-out code, should not reach production.

File: js/cruftfiles.js [hunk 4]
```diff
@@ -33,29 +33,70 @@ jQuery(document).ready(function($) {
...
+        });
+        $('.cruftfile-check').click(function() {
+          $('.cruftfile-select-all').each(function() {
+            this.checked = false;
+          });
+        });
         $( '#cruftfiles_status_loading img' ).fadeOut
         $('.cruftfile-delete-button').click(function(event) {
           event.preventDefault();
           var is_user_sure = confirm(seravo_cruftfiles_loc.confirm);
```
Comment (l3ku) [inline js/cruftfiles.js]: Please check that there are actually any files selected before attempting confirmation and sending the AJAX request to the server. If I don't choose any files and click "Delete", the confirmation message appears. After that if I press "OK" in the dialog, an error is caused to the browser console:
screen shot 2018-06-28 at 9 57 03

File: js/cruftfiles.js [hunk 5]
```diff
@@ -33,29 +33,70 @@ jQuery(document).ready(function($) {
...
+              }, 600, function() {
+                row.remove();
+                if ( $('#cruftfiles_entries').children().length == 0 ) {
+                  $( '#cruftfiles_status' ).children().remove();
+                  $( '#cruftfiles_status' ).append('<b>' + seravo_cruftfiles_loc.no_cruftfiles + '</b>');
+                }
+              });
             });
+
+            // if empty : show this
```
Comment (l3ku) [inline js/cruftfiles.js]: Show what?

File: js/cruftfiles.js (hunk 1 above)
Comment (mikkoaf) [reply js/cruftfiles.js]: We came to the conclusion the following conclusion for cases with less than 30 items considered as cruft the lower select all -checkbox is hidden and we removed the upper Delete button altogether.

image

We also improved the way how the checkboxes work when selecting and deselecting items. Now if all items are checked, the select all -boxes are checked as well.

**Pull request merged**